_posts/2016-11-29-future.md

```diff
@@ -0,0 +1,215 @@
+
+There is a lot of confusion around how plugins/presets interact regarding ordering. This results in bugs and issues with the config which require users to place plugins before/after others in a non-intuitive way.
+
+This is a fundamental change to the core of Babel which will take some work to figure out.
+
+### TypeScript Support
+
+Be able to understand TypeScript syntax (like we do with Flow) and add a plugin to strip TypeScript types for better interop. This would just mean parsing TypeScript specific syntax and stripping it out. Anything more would require a lot more work and discussion.
+
+### Use Type Information
+
+Integrate with a type system such as Flow/TypeScript to make optimizations. This just means that Babel should be able to gain the knowledge through those tools that an identifier `arr` is actually an `Array` or not.
+
+> There are a few plugins that actually do type checking: https://github.com/codemix/babel-plugin-typecheck and https://github.com/gcanti/babel-plugin-tcomb
+
+### Take in a Dependency Graph / Operate Multifile
+
+Then we could integrate with other tools like Webpack better. This would allow cross file transformations or whole codebase optimizations. The main reason for this would be for the minifier (being able to remove properties based on checking the usage across the whole application) or for example providing errors for missing/invalid import/exports.
+
+### Parser Errors
+
+Better error messages from the parser like in [Compiler Errors for Humans](http://elm-lang.org/blog/compiler-errors-for-humans).
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">babel-eslint@7.1.1: now adds the code frame when there&#39;s a parser error! <a href="https://t.co/yoxRpGXq5E">pic.twitter.com/yoxRpGXq5E</a></p>&mdash; Henry Zhu (@left_pad) <a href="https://twitter.com/left_pad/status/799388723896946692">November 17, 2016</a></blockquote>
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+> It is obvious that we all want to see helpful errors!
+
+- [babel/babylon#125](https://github.com/babel/babylon/pull/125): Instead of "Unexpected token" explain that `await` needs to be used in an async function.
+- Do better inference/guessing on user intention to prevent vague errors.
+
+### `babel-init`
+
+Basically a way to setup Babel easier like create-react-app does.
+
+- Setup a `.babelrc` from scratch
+- Maybe reuse the `babel` npm package to default to use babel-preset-env
+
+### Run [tc39/test262](https://github.com/tc39/test262)
+
+Running official spec tests against Babel to make sure we comply to the spec or at least know when we don't.
+
+### Smoke Tests
+
+It would be useful to either have a reverse [Greenkeeper](https://greenkeeper.io) or to run tests with master branch of Babel so that we can catch any major regressions before any release. In theory we would want to take the biggest open source projects that use Babel and run their tests.
```

Comment:
node does this with project canary https://github.com/nodejs/citgm. Might be nice to mention that next to `reverse greenkeeper`